src/main/java/graphql/schema/DataFetchingEnvironmentImpl.java

```diff
@@ -166,6 +166,11 @@ public ExecutionStepInfo getExecutionStepInfo() {
         return dataLoaderRegistry.getDataLoader(dataLoaderName);
     }
 
+    @Override
+    public DataLoaderRegistry getDataLoaderRegistry() {
+        return dataLoaderRegistry;
+    }
+
```

Comment:
I did this because we want to eventually get rid of ExecutionContext inside the DFE - so lets just get straight to the objects instead

Reply:
Interesting, what's the purpose to getting rid of the ExecutionContext?

Reply:
This is half wish in that the ExecutionContext really should have been an @Internal class.

For example it contans the "ExecutionStrategy"s to be used.  But that is not something that call back code like DataFecters should care about

It currently contains

```
    private final GraphQLSchema graphQLSchema;
    private final ExecutionId executionId;
    private final InstrumentationState instrumentationState;
    private final ExecutionStrategy queryStrategy;
    private final ExecutionStrategy mutationStrategy;
    private final ExecutionStrategy subscriptionStrategy;
    private final Map<String, FragmentDefinition> fragmentsByName;
    private final OperationDefinition operationDefinition;
    private final Document document;
    private final Map<String, Object> variables;
    private final Object root;
    private final Object context;
    private final Instrumentation instrumentation;
    private final List<GraphQLError> errors = new CopyOnWriteArrayList<>();
    private final Set<ExecutionPath> errorPaths = new HashSet<>();
    private final DataLoaderRegistry dataLoaderRegistry;
    private final CacheControl cacheControl;
    private final DeferSupport deferSupport = new DeferSupport();

```

There are useful things there but we regret making it all available as API

Reply:
I see, add the useful things directly now makes more sense vs opening everything.